pkg/receive/writer_test.go

```diff
@@ -174,3 +175,80 @@ func TestWriter(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkWriterTimeSeriesWithSingleLabel_10(b *testing.B)   { benchmarkWriter(b, 1, 10) }
+func BenchmarkWriterTimeSeriesWithSingleLabel_100(b *testing.B)  { benchmarkWriter(b, 1, 100) }
+func BenchmarkWriterTimeSeriesWithSingleLabel_1000(b *testing.B) { benchmarkWriter(b, 1, 1000) }
+
+func BenchmarkWriterTimeSeriesWith10Labels_10(b *testing.B)   { benchmarkWriter(b, 10, 10) }
+func BenchmarkWriterTimeSeriesWith10Labels_100(b *testing.B)  { benchmarkWriter(b, 10, 100) }
+func BenchmarkWriterTimeSeriesWith10Labels_1000(b *testing.B) { benchmarkWriter(b, 10, 1000) }
+
+func benchmarkWriter(b *testing.B, labelsNum int, seriesNum int) {
+	dir, err := ioutil.TempDir("", "test")
```

Comment:
To allow concurrent execution, maybe we should give the temp dir a unique name?
```suggestion
	dir, err := ioutil.TempDir("", "bench-writer")
```